### PR TITLE
[confighttp] Add `read_timeout` and `read_header_timeout` options to confighttp

### LINKED ISSUE #5699: Add `ReadHeaderTimeout` values
The following lint errors started showing up after the latest upgrade to golangci-lint:

```
golangci-lint run
config/confighttp/confighttp.go:321:10: G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (gosec)
        return &http.Server{
                Handler: handler,
        }, nil
config/confighttp/compression_test.go:106:12: G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (gosec)
                        srv := &http.Server{
                                Handler: handler,
                        }
config/confighttp/compression_test.go:204:12: G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (gosec)
                        srv := &http.Server{
                                Handler: httpContentDecompressor(handler),
                        }
make: *** [lint] Error 1
```

### PR DESCRIPTION
**Pull request by atoulme**
…

#### Description
Add `read_timeout` and `read_header_timeout` options to confighttp, to use as HTTP server config options.

#### Link to tracking issue
Fixes #5699

#### Documentation
Added to README.

### CODE REVIEW

**Comment by atoulme** [comment]
This is on hold, after merging https://github.com/open-telemetry/opentelemetry-collector/pull/10276

**Review by jpkrohling — APPROVED** [review]

**Comment by atoulme** [comment]
@jpkrohling rebased and ready to go.

File: service/internal/proctelemetry/config.go
Comment (jpkrohling) [inline service/internal/proctelemetry/config.go:108]: Just to double-check. this is unrelated to the change above, right?

**Review by jpkrohling — APPROVED** [review]

File: service/internal/proctelemetry/config.go
Comment (atoulme) [reply service/internal/proctelemetry/config.go:108]: it's because I also removed the gosec exception. The linter picked up the fact we didn't set this on the prometheus server as well.

**Comment by jpkrohling** [comment]
There are a few test failures related to this change. Can you address them?

**Comment by atoulme** [comment]
> There are a few test failures related to this change. Can you address them?

I am not sure how - this new config field clashes with an existing config field from contrib, and performs the same function as that field. My thinking is that we'd merge this, and update contrib and change it to use the new field. WDYT?

**Comment by jpkrohling** [comment]
Right, the failures are in contrib-tests, I missed that. LGTM then.

**Pull request closed without merging**
